src/ReachSets/AbstractReachSet.jl

```diff
@@ -214,6 +214,15 @@ by `t0`.
 """
 function shift(R::AbstractReachSet, t0::Number) end
 
+# --------------
+# Other methods
+# --------------
+
+LazySets.constraints_list(R::AbstractReachSet) = constraints_list(set(R))
```

Comment:
Not sure about this; I think of `AbstractReachSet` as a very general representation, with no particular link to `LazySets`. OTOH, `AbstractLazyReachSet` has some useful fallbacks (eg. [here](https://github.com/JuliaReach/LazySets.jl/blob/master/src/Interfaces/LazySet.jl#L1134)).

If you needed to apply a method for TM reach-sets, we can dispatch on `TaylorModelReachSet` directly or `AbstractTaylorModelReachSet`.

Reply:
I checked and we currently have such methods [here](https://github.com/JuliaReach/ReachabilityAnalysis.jl/blob/master/src/ReachSets/TaylorModelReachSet.jl#L111). The underlying problem is that `UnionSet` and `UnionSetArray` are not yet `<:LazySet`. I would suggest to do

```julia
for ST in (LazySet, UnionSet, UnionSetArray)
    @eval Base.:⊆(R::TaylorModelReachSet, X::$(ST)) = ⊆(overapproximate(set(R), Zonoope), X))
    @eval Base.:⊆(X::$(ST), R::TaylorModelReachSet) = ⊆(X, overapproximate(set(R), Zonotope))
end
```
like in your changes below (but we need the call to `overapproximate`, since LazySets doesn't reason with the TM)

Reply:
> Not sure about this; I think of `AbstractReachSet` as a very general representation, with no particular link to `LazySets`.

Okay, this was not clear to me. I thought that `AbstractLazyReachSet` is just lazy.

Reply:
👍  it's perhaps not the best name, but .. 

then you make the proposed change or shall i?

Reply:
Done. It works without the TMRS (I checked).